<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A1 Basic Bioconductor resources for working with the T2T reference genome â€¢ BiocT2T</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A1 Basic Bioconductor resources for working with the T2T reference genome">
<meta property="og:description" content="BiocT2T">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocT2T</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A1_T2Toverview.html">A1 Basic Bioconductor resources for working with the T2T reference genome</a>
    </li>
    <li>
      <a href="../articles/B1_1KG_chr17.html">B1 Acquiring and exploring 3202 1KG genotypes from chr17 with T2T reference</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A1 Basic Bioconductor resources for working with the T2T reference genome</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">April 04, 2023</h4>
      
      
      <div class="hidden name"><code>A1_T2Toverview.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="objectives">Objectives<a class="anchor" aria-label="anchor" href="#objectives"></a>
</h2>
<p>This package demonstrates the use of genotypes obtained via calls against the T2T reference genome. Our objectives are:</p>
<ul>
<li>acquire and make the genotypes accessible to R/Bioconductor</li>
<li>bind relevant sample-level metadata to the resource</li>
<li>acquire associated RNA-seq assay outputs for a subset of the genotyped samples</li>
<li>perform a series of GWAS to identify expression-based quantitative trait loci (eQTL)</li>
</ul>
</div>
<div class="section level2">
<h2 id="the-reference-genomic-sequence">The reference genomic sequence<a class="anchor" aria-label="anchor" href="#the-reference-genomic-sequence"></a>
</h2>
<div class="section level3">
<h3 id="grch38">GRCh38<a class="anchor" aria-label="anchor" href="#grch38"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.NCBI.GRCh38</span><span class="op">)</span></span>
<span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">BSgenome.Hsapiens.NCBI.GRCh38</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Seqinfo object with 455 sequences (1 circular) from GRCh38 genome:</span></span>
<span><span class="co">##   seqnames                       seqlengths isCircular genome</span></span>
<span><span class="co">##   1                               248956422      FALSE GRCh38</span></span>
<span><span class="co">##   2                               242193529      FALSE GRCh38</span></span>
<span><span class="co">##   3                               198295559      FALSE GRCh38</span></span>
<span><span class="co">##   4                               190214555      FALSE GRCh38</span></span>
<span><span class="co">##   5                               181538259      FALSE GRCh38</span></span>
<span><span class="co">##   ...                                   ...        ...    ...</span></span>
<span><span class="co">##   HSCHR19KIR_FH08_BAX_HAP_CTG3_1     200773      FALSE GRCh38</span></span>
<span><span class="co">##   HSCHR19KIR_FH13_A_HAP_CTG3_1       170148      FALSE GRCh38</span></span>
<span><span class="co">##   HSCHR19KIR_FH13_BA2_HAP_CTG3_1     215732      FALSE GRCh38</span></span>
<span><span class="co">##   HSCHR19KIR_FH15_A_HAP_CTG3_1       170537      FALSE GRCh38</span></span>
<span><span class="co">##   HSCHR19KIR_RP5_B_HAP_CTG3_1        177381      FALSE GRCh38</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BSgenome.Hsapiens.NCBI.GRCh38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.NCBI.GRCh38/man/package.html" class="external-link">Hsapiens</a></span></span></code></pre></div>
<pre><code><span><span class="co">## | BSgenome object for Human</span></span>
<span><span class="co">## | - organism: Homo sapiens</span></span>
<span><span class="co">## | - provider: NCBI</span></span>
<span><span class="co">## | - genome: GRCh38</span></span>
<span><span class="co">## | - release date: 2013-12-17</span></span>
<span><span class="co">## | - 455 sequence(s):</span></span>
<span><span class="co">## |     1                                  2                                 </span></span>
<span><span class="co">## |     3                                  4                                 </span></span>
<span><span class="co">## |     5                                  6                                 </span></span>
<span><span class="co">## |     7                                  8                                 </span></span>
<span><span class="co">## |     9                                  10                                </span></span>
<span><span class="co">## |     ...                                ...                               </span></span>
<span><span class="co">## |     HSCHR19KIR_FH05_B_HAP_CTG3_1       HSCHR19KIR_FH06_A_HAP_CTG3_1      </span></span>
<span><span class="co">## |     HSCHR19KIR_FH06_BA1_HAP_CTG3_1     HSCHR19KIR_FH08_A_HAP_CTG3_1      </span></span>
<span><span class="co">## |     HSCHR19KIR_FH08_BAX_HAP_CTG3_1     HSCHR19KIR_FH13_A_HAP_CTG3_1      </span></span>
<span><span class="co">## |     HSCHR19KIR_FH13_BA2_HAP_CTG3_1     HSCHR19KIR_FH15_A_HAP_CTG3_1      </span></span>
<span><span class="co">## |     HSCHR19KIR_RP5_B_HAP_CTG3_1                                          </span></span>
<span><span class="co">## | </span></span>
<span><span class="co">## | Tips: call 'seqnames()' on the object to get all the sequence names, call</span></span>
<span><span class="co">## | 'seqinfo()' to get the full sequence info, use the '$' or '[[' operator to</span></span>
<span><span class="co">## | access a given sequence, see '?BSgenome' for more information.</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hs38</span> <span class="op">=</span> <span class="va">BSgenome.Hsapiens.NCBI.GRCh38</span></span>
<span><span class="va">hs38</span><span class="op">$</span><span class="va">`1`</span></span></code></pre></div>
<pre><code><span><span class="co">## 248956422-letter DNAString object</span></span>
<span><span class="co">## seq: <span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>...<span style="color: #EEEEEE; background-color: #B2B2B2;">NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="t2t">T2T<a class="anchor" aria-label="anchor" href="#t2t"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0</span><span class="op">)</span></span>
<span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Seqinfo object with 25 sequences (1 circular) from T2T-CHM13v2.0 genome:</span></span>
<span><span class="co">##   seqnames seqlengths isCircular        genome</span></span>
<span><span class="co">##   1         248387328      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   2         242696752      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   3         201105948      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   4         193574945      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   5         182045439      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   ...             ...        ...           ...</span></span>
<span><span class="co">##   21         45090682      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   22         51324926      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   X         154259566      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   Y          62460029      FALSE T2T-CHM13v2.0</span></span>
<span><span class="co">##   MT            16569       TRUE T2T-CHM13v2.0</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0/man/package.html" class="external-link">Hsapiens</a></span></span></code></pre></div>
<pre><code><span><span class="co">## | BSgenome object for Human</span></span>
<span><span class="co">## | - organism: Homo sapiens</span></span>
<span><span class="co">## | - provider: NCBI</span></span>
<span><span class="co">## | - genome: T2T-CHM13v2.0</span></span>
<span><span class="co">## | - release date: 2022/01/24</span></span>
<span><span class="co">## | - 25 sequence(s):</span></span>
<span><span class="co">## |     1   2   3   4   5   6   7   8   9   10  11  12  13  14  15  16  17  18 </span></span>
<span><span class="co">## |     19  20  21  22  X   Y   MT                                             </span></span>
<span><span class="co">## | </span></span>
<span><span class="co">## | Tips: call 'seqnames()' on the object to get all the sequence names, call</span></span>
<span><span class="co">## | 'seqinfo()' to get the full sequence info, use the '$' or '[[' operator to</span></span>
<span><span class="co">## | access a given sequence, see '?BSgenome' for more information.</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hst2t</span> <span class="op">=</span> <span class="va">BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0</span></span>
<span><span class="va">hst2t</span><span class="op">$</span><span class="va">`1`</span></span></code></pre></div>
<pre><code><span><span class="co">## 248387328-letter DNAString object</span></span>
<span><span class="co">## seq: <span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AAA</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CCCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #AFFFAF;">CC</span>...<span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">GGG</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span></span></code></pre>
<p>18 million non-called positions in GRCh38 are now called.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">alphabetFrequency</span><span class="op">(</span><span class="va">hst2t</span><span class="op">$</span><span class="va">`1`</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        A        C        G        T        M        R        W        S </span></span>
<span><span class="co">## 73600418 51017014 52064401 71705495        0        0        0        0 </span></span>
<span><span class="co">##        Y        K        V        H        D        B        N        - </span></span>
<span><span class="co">##        0        0        0        0        0        0        0        0 </span></span>
<span><span class="co">##        +        . </span></span>
<span><span class="co">##        0        0</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">alphabetFrequency</span><span class="op">(</span><span class="va">hs38</span><span class="op">$</span><span class="va">`1`</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        A        C        G        T        M        R        W        S </span></span>
<span><span class="co">## 67070277 48055043 48111528 67244164        1        1        0        0 </span></span>
<span><span class="co">##        Y        K        V        H        D        B        N        - </span></span>
<span><span class="co">##        0        0        0        0        0        0 18475408        0 </span></span>
<span><span class="co">##        +        . </span></span>
<span><span class="co">##        0        0</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="genes-and-transcripts">Genes and transcripts<a class="anchor" aria-label="anchor" href="#genes-and-transcripts"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocT2T</span><span class="fu">::</span><span class="fu"><a href="../reference/install_early_t2t_txdb.html">install_early_t2t_txdb</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.NCBI.CHM13v2</span><span class="op">)</span></span>
<span><span class="va">TxDb.Hsapiens.NCBI.CHM13v2</span></span></code></pre></div>
<pre><code><span><span class="co">## TxDb object:</span></span>
<span><span class="co">## # Db type: TxDb</span></span>
<span><span class="co">## # Supporting package: GenomicFeatures</span></span>
<span><span class="co">## # Data source: chm13v2.0_RefSeq_Liftoff_v4.gff3</span></span>
<span><span class="co">## # Organism: Homo sapiens</span></span>
<span><span class="co">## # Taxonomy ID: 9606</span></span>
<span><span class="co">## # miRBase build ID: NA</span></span>
<span><span class="co">## # method: Liftoff_v4</span></span>
<span><span class="co">## # source: JHU</span></span>
<span><span class="co">## # Resource URL: https://ccb.jhu.edu/T2T.shtml</span></span>
<span><span class="co">## # Genome: T2T-CHM13v2.0</span></span>
<span><span class="co">## # Nb of transcripts: 181747</span></span>
<span><span class="co">## # Db created by: GenomicFeatures package from Bioconductor</span></span>
<span><span class="co">## # Creation time: 2023-03-16 11:00:14 -0400 (Thu, 16 Mar 2023)</span></span>
<span><span class="co">## # GenomicFeatures version at creation time: 1.51.4</span></span>
<span><span class="co">## # RSQLite version at creation time: 2.3.0</span></span>
<span><span class="co">## # DBSCHEMAVERSION: 1.2</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">txdb</span> <span class="op">=</span> <span class="va">TxDb.Hsapiens.NCBI.CHM13v2</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 45316 ranges and 1 metadata column:</span></span>
<span><span class="co">##              seqnames            ranges strand |     gene_id</span></span>
<span><span class="co">##                 &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##         A1BG    chr19 61441599-61449907      - |        A1BG</span></span>
<span><span class="co">##     A1BG-AS1    chr19 61448385-61451599      + |    A1BG-AS1</span></span>
<span><span class="co">##         A1CF    chr10 51648044-51734261      - |        A1CF</span></span>
<span><span class="co">##          A2M    chr12   9049893-9098416      - |         A2M</span></span>
<span><span class="co">##      A2M-AS1    chr12   9047362-9050240      + |     A2M-AS1</span></span>
<span><span class="co">##          ...      ...               ...    ... .         ...</span></span>
<span><span class="co">##   gene-TRNS2     chrM       11630-11688      + |  gene-TRNS2</span></span>
<span><span class="co">##    gene-TRNT     chrM       15311-15376      + |   gene-TRNT</span></span>
<span><span class="co">##    gene-TRNV     chrM         1026-1094      + |   gene-TRNV</span></span>
<span><span class="co">##    gene-TRNW     chrM         4935-5002      + |   gene-TRNW</span></span>
<span><span class="co">##    gene-TRNY     chrM         5249-5314      - |   gene-TRNY</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences (1 circular) from T2T-CHM13v2.0 genome; no seqlengths</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 181747</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">exons</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 374407</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vincent Carey.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
