<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A2 Acquiring 3202 1KG genotypes from chr17 with T2T reference â€¢ BiocT2T</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="A2 Acquiring 3202 1KG genotypes from chr17 with T2T reference">
<meta property="og:description" content="BiocT2T">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocT2T</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/A1_T2Toverview.html">A1 Basic Bioconductor resources for working with the T2T reference genome</a>
    </li>
    <li>
      <a href="../articles/A2_1KG_chr17.html">A2 Acquiring 3202 1KG genotypes from chr17 with T2T reference</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>A2 Acquiring 3202 1KG genotypes from chr17 with T2T reference</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">July 23, 2023</h4>
      
      
      <div class="hidden name"><code>A2_1KG_chr17.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="objectives">Objectives<a class="anchor" aria-label="anchor" href="#objectives"></a>
</h2>
<p>This package demonstrates the use of genotypes obtained via calls against the T2T reference genome. Our objectives are:</p>
<ul>
<li>acquire and make the genotypes accessible to R/Bioconductor</li>
<li>bind relevant sample-level metadata to the resource</li>
<li>acquire associated RNA-seq assay outputs for a subset of the genotyped samples</li>
<li>perform a series of GWAS to identify expression-based quantitative trait loci (eQTL)</li>
</ul>
</div>
<div class="section level2">
<h2 id="acquiring-3202-human-genotypes-called-against-the-t2t-reference">Acquiring 3202 human genotypes called against the T2T reference<a class="anchor" aria-label="anchor" href="#acquiring-3202-human-genotypes-called-against-the-t2t-reference"></a>
</h2>
<div class="section level3">
<h3 id="the-cloud-resource-environment-variable-setting">The cloud resource; environment variable setting<a class="anchor" aria-label="anchor" href="#the-cloud-resource-environment-variable-setting"></a>
</h3>
<p>We have made a large resource available in the hail.is MatrixTable format. This can be retrieved without egress charges using the code in this vignette. A convenient approach uses rclone, as described in the appendix. The zipped MatrixTable folders are at</p>
<pre><code>https://mghp.osn.xsede.org/bir190004-bucket01/Bioc1KGt2t/t17.zip</code></pre>
<p>This 42GB file should be unzipped in a folder to which the environment variable <code>HAIL_T2T_CHR17</code> will point.</p>
</div>
<div class="section level3">
<h3 id="initial-checks-of-the-resource">Initial checks of the resource<a class="anchor" aria-label="anchor" href="#initial-checks-of-the-resource"></a>
</h3>
<p>Ensure that <code>Sys.getenv("HAIL_T2T_CHR17")</code> returns the location of the unzipped MatrixTable resource.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BiocT2T</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/BiocHail" class="external-link">BiocHail</a></span><span class="op">)</span></span>
<span><span class="va">hl</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocHail/man/hail_init.html" class="external-link">hail_init</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># NB the following two commands are now encapsulated in the rg_update function</span></span>
<span><span class="va">nn</span> <span class="op">&lt;-</span> <span class="va">hl</span><span class="op">$</span><span class="fu">get_reference</span><span class="op">(</span><span class="st">'GRCh38'</span><span class="op">)</span></span>
<span><span class="va">nn</span> <span class="op">&lt;-</span> <span class="va">nn</span><span class="op">$</span><span class="fu">read</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"json/t2tAnVIL.json"</span>, package<span class="op">=</span><span class="st">"BiocHail"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># updates the hail reference genome</span></span>
<span><span class="va">bigloc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"HAIL_T2T_CHR17"</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">bigloc</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">mt17</span> <span class="op">&lt;-</span> <span class="va">hl</span><span class="op">$</span><span class="fu">read_matrix_table</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"HAIL_T2T_CHR17"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">mt17</span><span class="op">$</span><span class="fu">count</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span>
<span><span class="co">## [1] 3824434</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[2]]</span></span>
<span><span class="co">## [1] 3202</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="population-stratification-assessment-via-pca">Population stratification assessment via PCA<a class="anchor" aria-label="anchor" href="#population-stratification-assessment-via-pca"></a>
</h2>
<p>The following computation takes close to 30 minutes with 64 cores. We saved the loadings in <code>pc_990kloci</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcastuff</span> <span class="op">=</span> <span class="va">hl</span><span class="op">$</span><span class="fu">hwe_normalized_pca</span><span class="op">(</span><span class="va">mt17</span><span class="op">$</span><span class="fu">sample_rows</span><span class="op">(</span><span class="fl">.25</span><span class="op">)</span><span class="op">$</span><span class="va">GT</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pc_990kloci</span><span class="op">)</span></span>
<span><span class="va">pc_990kloci</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                PC1         PC2         PC3          PC4</span></span>
<span><span class="co">## HG00096 0.08571635 -0.09670515 -0.04432893  0.032206907</span></span>
<span><span class="co">## HG00097 0.07130168 -0.06820003 -0.01072185  0.004602563</span></span>
<span><span class="co">## HG00099 0.06578964 -0.08644191 -0.01452058 -0.010706626</span></span></code></pre>
<p>We expect these PCs to provide information on population of origin. Metadata about sample origins is available in BiocHail.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocHail/man/path_1kg_annotations.html" class="external-link">path_1kg_annotations</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">anp</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="co"># library(dplyr) # Available via Depends</span></span>
<span><span class="va">pcdf</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">pc_990kloci</span><span class="op">)</span></span>
<span><span class="va">pcdf</span><span class="op">$</span><span class="va">Sample</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pcdf</span><span class="op">)</span></span>
<span><span class="va">lj</span> <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">md</span>, <span class="va">pcdf</span>, by<span class="op">=</span><span class="st">"Sample"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">lj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 3500   16</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="appendix-using-rclone-with-docker-to-get-the-chr17-data">Appendix: Using rclone with docker to get the chr17 data<a class="anchor" aria-label="anchor" href="#appendix-using-rclone-with-docker-to-get-the-chr17-data"></a>
</h2>
<p>It can be painful to install and configure rclone. We use a docker container. Let RC_DATADIR be an environment variable evaluating to an available folder.</p>
<p>Also, place the text file with contents</p>
<pre><code>[osn]
type = s3
provider = AWS
endpoint = https://mghp.osn.xsede.org
acl = public
no_check_bucket = true</code></pre>
<p>in a file <code>rclone.conf</code> in a folder pointed to by the environment variable <code>RC_CONFDIR</code>.</p>
<p>Then the following</p>
<pre><code>docker run -v $RC_DATADIR:/data -v $RC_CONFDIR:/config/rclone -t rclone/rclone:latest ls osn:/bir190004-bucket01/Bioc1KGt2t</code></pre>
<p>will list the files with 1KG samples genotyped against the T2T reference.</p>
<p>Use the rclone <code>copyto</code> command to obtain a local copy of the zip file <code>t17.zip</code> in the folder pointed to by <code>$RC_DATADIR</code>:</p>
<pre><code>docker run -v $RC_DATADIR:/data -v $RC_CONFDIR:/config/rclone -t rclone/rclone:latest copyto osn:/bir190004-bucket01/Bioc1KGt2t/t17.zip ./t17.zip</code></pre>
<p>This file should be unzipped in a folder to which the environment variable <code>HAIL_T2T_CHR17</code> will point.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vincent Carey.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
